### PR TITLE
Use JSON schema

### PR DESCRIPTION
**Pull request by sbrunner**
The goal of USAGE.rst is to be able to add it in the c2cgeoportal documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,763 +93,54 @@ file](https://github.com/camptocamp/tilecloud-chain/blob/master/tilecloud_chain/
...
-``` {.sourceCode .bash}
+```{.sourceCode .bash}
 docker-compose -p tilecloud exec test python setup.py nosetests --logging-filter=tilecloud,tilecloud_chain --attr '!'nopy3
 ```
 
-Environment variables
----------------------
+## Documentation
+
+As documatation you can read the `tilecloud_chain/USAGE.rst`.
```
Comment (arnaud-morvan) [inline README.md:132]:
```suggestion
As documentation you can read the `tilecloud_chain/USAGE.rst`.
```

Comment (arnaud-morvan) [inline README.md:132]: May use absolute link, not sure a relative link will be correctly expanded in PyPi ?

File: tilecloud_chain/tests/tilegeneration/test.yaml [hunk 2]
```diff
@@ -63,10 +63,8 @@ caches:
         folder: tiles
 
 defaults:
-    layer: &layer
+    all_layer: &all_layer
```
Comment (arnaud-morvan) [inline tilecloud_chain/tests/tilegeneration/test.yaml:66]: I would use different naming :
- `all_layer` => `layer` or `base_layer`
- `layer` => `mapserver_layer`

`all_layer` make no sense (`one_layer` or `all_layers`, but cannot be `all_layer`).

File: tilecloud_chain/__init__.py [hunk 3]
```diff
@@ -304,56 +306,17 @@ def __init__(
...
-        except NotMappingError as e:  # pragma: no cover
-            logger.error("The config file '%s' is invalid.\n - %s", config_file, e.msg)
+        errors = sorted(
+            [
+                f' - {".".join([str(i) for i in e.path] if e.path else "/")}: {e.message}'
+                for e in validator.iter_errors(self.config)
+            ]
+        )
+        if errors:
+            logger.error("The config file '%s' is invalid.\n%s", config_file, "\n".join(errors))
```
Comment (arnaud-morvan) [inline tilecloud_chain/__init__.py:319]: May create a `validate` method and call it here.

**Review by arnaud-morvan — APPROVED** [review]
Minor comments

File: README.md (hunk 1 above)
Comment (sbrunner) [reply README.md:132]: +1 :-)

**Pull request merged**